invenio_rdm_records/collections/api.py

```diff
@@ -91,8 +91,21 @@ def resolve(cls, *, id_=None, slug=None, ctree_id=None, depth=2):
     @classmethod
     def resolve_many(cls, ids_=None, depth=2):
         """Resolve many collections by ID."""
-        _ids = ids_ or []
-        return [cls(c, depth) for c in cls.model_cls.read_many(_ids)]
+        if not ids_:
+            return []
+        return [cls(c, depth) for c in cls.model_cls.read_many(ids_)]
+
+    @classmethod
+    def resolve_all(cls, depth=2):
```

Comment:
Any reason why it is called `resolve_all` and/or `read_all` between service, api and models?

Reply:
Yes, this was done explicitly to separate both.  I expect that  'read` and `resolve` to return very different things (service result item X API object ). 

But re-reading the code we have in other places, `resolve` seems to be used more for resolvers (e.g., entity resolvers), systemfields, and `pid` (where the service method is also named `resolve`). Since we're not in that situation here, I think it can be modified to `read*` all over.